scripts/add_electricity.py

```diff
@@ -331,17 +331,20 @@ def attach_conventional_generators(n, costs, ppl, conventional_carriers, extenda
         
         # Generators with technology affected
         idx = n.generators.query("carrier == @carrier").index
-        factors = conventional_config[carrier].get("energy_availability_factors")
-
-        if isinstance(factors, float):
-            # Single value affecting all generators of technology k indiscriminantely of country
-            n.generators.loc[idx, "p_max_pu"] = factors
-        elif isinstance(factors, str):
-            factors = pd.read_csv(factors, index_col=0)['factor']
-            # Values affecting generators of technology k country-specific
-            # First map generator buses to countries; then map countries to p_max_pu
-            bus_factors = n.buses.country.map(factors)
-            n.generators.p_max_pu.update(n.generators.loc[idx].bus.map(bus_factors).dropna())
+
+        for key in list(set(conventional_carriers[carrier]) & set(n.generators)):
```

Comment:
does that work? `conventional_carriers` is a list of carriers, `carrier` is a `str`, so what is `conventional_carriers[carrier]`?

Reply:
also, n.generators is a `pd.Dataframe`, maybe you meant `n.generators.carrier`?

why not toe following? (I'm not sure what you're trying to filter)
```suggestion
        n.generators.loc[idx].carrier.unique():
```

Reply:
ahm, you probably wanted to check if the key given in config is in generator columns?
```suggestion
        for key in list(set(conventional_config[carrier]) & set(n.generators)):
```

Reply:
you are just right